tests/helpers/k8s.bash

```diff
@@ -628,3 +629,166 @@ function kindbox_cluster_teardown() {
   # Delete cluster config.
   rm -rf /root/.kube/${cluster}-config
 }
+
+
+################################################################################
+# K3s specific functions
+################################################################################
+
+
+function k3s_all_nodes_ready() {
```

Comment:
Question: don't see anything specific to k3s in this function ... seems like it would work as well for K8s ...

Reply:
Not exactly the same function -- worker-id offset is different and workers naming convention is different too among KinD, Kindbox and K3s setups.

Reply:
mmm ... why? why would we change the worker-id offset and naming convention? If we keep it the same then we can reuse the helpers functions more easily ...

Reply:
That has always been the case, that's why we have separate `k8s_all_nodes_ready` functions since day one to differentiate between KinD and Kinbox.

Reply:
A couple of points:

1) I don't see any consumers of "kind_all_nodes_ready" (because our tests use kindbox).
2) I think it's OK to have "k3s_all_nodes_ready", or "kind_all_nodes_ready", but not both because in theory they should be identical.

Reply:
If we ever add KinD tests again, which it's just a matter of time given the importance of this use-case for us, this `kind_all_nodes_ready` will be handy to have, that's why it hasn't been removed.

AFIAK, we have no control over the naming-convention utilized by KinD (embedded in their binary), which is different from the one we use at Kindbox and the one utilized by K3s. KinD makes use of a somewhat unintuitive convention (IMO) due to historical reasons (e.g. first node has no index, second node makes use of index-2, etc). That's different from what we do at Kindbox and also at K3s.

Reply:
Ok got it ...  I don't see us using the `kind_all_nodes_ready` function because it would require that KinD support sysbox as a runtime which is currently not under our control. I would remove it.